### ISSUE TITLE
[SEVERE ERROR] Alpha17 Invalid Player Data upon entering world

### ISSUE THREAD

**Issue by Shibva** [DaFuqs/Spectrum#222]
**Describe the bug**
```
[00:14:44] [Server thread/ERROR]: Couldn't place player in world
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_1767.toString()" because "this.color" is null
	at Not Enough Crashes deobfuscated stack trace.(1.19.2+build.28) ~[?:?]
	at de.dafuqs.spectrum.entity.type_specific_predicates.ShulkerPredicate.typeSpecificToJson(ShulkerPredicate.java:36) ~[spectrum-1.7.0-deeper-down-alpha17.jar:?]
	at net.minecraft.predicate.entity.TypeSpecificPredicate.toJson(TypeSpecificPredicate:88) ~[?:?]
	at net.minecraft.predicate.entity.EntityPredicate.toJson(EntityPredicate:207) ~[?:?]
	at net.minecraft.loot.condition.EntityPropertiesLootCondition$Serializer.toJson(EntityPropertiesLootCondition:58) ~[?:?]
	at net.minecraft.loot.condition.EntityPropertiesLootCondition$Serializer.toJson(EntityPropertiesLootCondition:55) ~[?:?]
	at net.minecraft.util.JsonSerializing$GsonSerializer.serialize(JsonSerializing:125) ~[?:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:81) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:69) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.write(ArrayTypeAdapter.java:95) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJson(Gson.java:735) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:628) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:607) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:155) ~[gson-2.8.9.jar:?]
	at net.minecraft.loot.LootPool$Serializer.serialize(LootPool:188) ~[?:?]
	at net.minecraft.loot.LootPool$Serializer.serialize(LootPool:169) ~[?:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:81) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:69) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.write(ArrayTypeAdapter.java:95) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJson(Gson.java:735) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:628) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:607) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:155) ~[gson-2.8.9.jar:?]
	at net.minecraft.loot.LootTable$Serializer.serialize(LootTable:240) ~[?:?]
	at net.minecraft.loot.LootTable$Serializer.serialize(LootTable:210) ~[?:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:81) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJson(Gson.java:735) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:628) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:607) ~[gson-2.8.9.jar:?]
	at uk.me.desert_island.rer.RoughlyEnoughResources.sendLootToPlayers(RoughlyEnoughResources.java:53) ~[rer-2.6.0.jar:?]
	at net.minecraft.server.PlayerManager.handler$ijp003$roughlyenoughresources$onPlayerConnect(PlayerManager:12506) ~[?:?]
	at net.minecraft.server.PlayerManager.onPlayerConnect(PlayerManager:289) ~[?:?]
	at net.minecraft.server.network.ServerLoginNetworkHandler.addToServer(ServerLoginNetworkHandler:151) ~[?:?]
	at net.minecraft.server.network.ServerLoginNetworkHandler.acceptPlayer(ServerLoginNetworkHandler:138) ~[?:?]
	at net.minecraft.server.network.ServerLoginNetworkHandler.redirect$eno000$fabric-networking-api-v1$handlePlayerJoin(ServerLoginNetworkHandler:563) ~[?:?]
	at net.minecraft.server.network.ServerLoginNetworkHandler.tick(ServerLoginNetworkHandler:73) ~[?:?]
	at net.minecraft.network.ClientConnection.tick(ClientConnection:248) ~[?:?]
	at net.minecraft.server.ServerNetworkIo.tick(ServerNetworkIo:183) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer:886) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer:819) ~[client-intermediary.jar:?]
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer:98) ~[?:?]
	at net.minecraft.server.MinecraftServer.handler$dam000$carpet$modifiedRunLoop(MinecraftServer:5385) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer:644) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer:257) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
[00:14:44] [Server thread/INFO]: Shibva_ lost connection: Invalid player data
```
TLDR: this.color is null; happens on existing and newly created worlds

**To Reproduce**
??? issue might have been absent in the workspace

**Mod version**
Which version of the mod are you currently using?
DeeperDown Alpha17

**Crash Report**
https://gist.github.com/Shibva/6f1398238c258c6818c9bad015cd0cb7

**Comment by Gronglegrowth** [comment]
This crash can be temporarily avoided by disabling Roughly Enough Resources; it also triggers when EMI Loot is present, which I and a few others had found out earlier. It has something to do with the loot tables of the shulkers I think, but that is just a guess on my part.

**Comment by Shibva** [comment]
@DaFuqs this issue will now be closed, Deeper Down Alpha 18 patched the issue sucessfuly
